src/drivers/imu/lsm303d/lsm303d.cpp

```diff
@@ -1656,8 +1659,17 @@ LSM303D::mag_measure()
 	rotate_3f(_rotation, xraw_f, yraw_f, zraw_f);
 
 	mag_report.x = ((xraw_f * _mag_range_scale) - _mag_scale.x_offset) * _mag_scale.x_scale;
+	mag_report.x += ((yraw_f * _mag_range_scale) - _mag_scale.y_offset) * _mag_scale.x_offdiag;
```

Comment:
This ends up being a lot of verbose duplicated code. Why don't we just start using Matrix?